### DISCUSSION TITLE
Extracting a signature matrix from Scaden

### DISCUSSION THREAD

**Discussion by nicolasgoedert97** [KevinMenden/scaden#71] [General]
Hey, I was wondering if it is possible to obtain a signature matrix from Scaden?

Other deconvolution algorithms, like Cibersort for example, tend to produce a signature matrix in order to deconvolute a bulk RNA sample into cell fractions. In the source code of scaden (precisely **scaden/preprocessing/bulk_simulation.py**) there is a method called: **generate_signature(x, y)**, which returns an object called **signature_matrix**, but appearently this function is never called by Scaden, or am I wrong?. 

My question would be: Is it possible to obtain a such matrix from Scaden?

**Comment by KevinMenden** [comment]
Hi,

good catch :) this was function was indeed used to generate signature matrices to use with Cibersert and the likes, in order to compare them to Scaden. The function calculate the average expression for each cell in the scRNA-seq dataset. Depending on the deconvolution algorithm, some additional processing is necessary. But usually they provide functionality for that.

The function is not used in Scaden anymore because Scaden doesn't require a signature matrix. If you think  that's useful though, I could make it available in the next version?
